plugin/metrics/prometheus/metricsstore/reader_test.go

```diff
@@ -316,17 +317,61 @@ func TestWarningResponse(t *testing.T) {
 
 func TestGetRoundTripper(t *testing.T) {
 	for _, tc := range []struct {
-		name       string
-		tlsEnabled bool
+		name                  string
+		tlsEnabled            bool
+		tokenFilePath         string
+		allowTokenFromContext bool
+		wantBearer            string
 	}{
-		{"tls tlsEnabled", true},
-		{"tls disabled", false},
+		{
+			"tls enabled with token from file and token from context is not considered ",
+			true,
+			"testdir/test_file.txt",
+			false,
+			"token from file",
+		},
+		{
+			"tls enabled with token from context",
+			true,
+			"",
+			true,
+			"token from context",
+		},
+		{
+			"tls enabled with token from file with allowTokenFromContext, should prefer using token from context",
+			true, "testdir/test_file.txt",
+			true,
+			"token from context",
+		},
+		{
+			"tls disabled with token from context",
+			false,
+			"",
+			true,
+			"token from context",
+		},
```

Comment:
I think we can make the test cases a bit more focussed and readable by separating the TLS  and bearer token concerns from test cases. What do you think of the following suggestion?

```suggestion
			name:       "tls enabled",
			tlsEnabled: true,
		},
		{
			name:       "tls disabled",
			tlsEnabled: false,
		},
		{
			name:          "token from file only",
			tokenFilePath: "testdir/test_file.txt",
			wantBearer:    "token from file",
		},
		{
			name:                  "propagate bearer token from context only",
			allowTokenFromContext: true,
			wantBearer:            "token from context",
		},
		{
			name:                  "token from file with bearer token propagation, should prefer using token from context",
			tokenFilePath:         "testdir/test_file.txt",
			allowTokenFromContext: true,
			wantBearer:            "token from context",
		},
```